geth/peers/peerpool.go

```diff
@@ -135,6 +132,9 @@ func (p *PeerPool) Start(server *p2p.Server) error {
 		p.wg.Done()
 	}()
 
+	// discovery must be already started when pool is started
+	signal.SendDiscoveryStarted()
```

Comment:
why move?

Reply:
I thought that may be an issue at the beginning. Is there any problem with calling it here when actually the whole setup is done?

Also, we may need to subscribe first and then start searching for topics. Otherwise, at least in theory, it may happen that we find a peer before subscribing to the server.